### PR TITLE
[opt](variable) force update some variable by variable version

### PR DESCRIPTION
**Pull request by morrySnow**
variable version:
000-100: doris-2.0.x
100-200: doris-2.1.x
200-300: doris-3.0.x

update variables

000:
nereids_timeout_second = 30 if original value is 5

100:
enable_nereids_dml = true
enable_nereids_dml_with_pipeline = true
enable_nereids_planner = true
enable_fallback_to_original_planner = true
enable_pipeline_x_engine = true

200:
enable_fallback_to_original_planner = false

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 1]
```diff
@@ -1599,6 +1599,47 @@ private void transferToMaster() {
                 // Set initial root password if master FE first time launch.
                 auth.setInitialRootPassword(Config.initial_root_password);
             } else {
+                int currentVariableVersion = VariableMgr.newSessionVariable().variableVersion;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:1602]: How about create a new class and move these logic to there.
The Env.java is too big

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -663,6 +671,9 @@ public class SessionVariable implements Serializable, Writable {
     public static final String ADAPTIVE_PIPELINE_TASK_SERIAL_READ_ON_LIMIT =
                                     "adaptive_pipeline_task_serial_read_on_limit";
 
+    @VariableMgr.VarAttr(name = VARIABLE_VERSION, flag = VariableMgr.INVISIBLE | VariableMgr.READ_ONLY)
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:674]: should move it to GlobalVariable?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java (hunk 1 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java]: done

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Comment by morrySnow** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/qe/VariableMgr.java [hunk 3]
```diff
@@ -947,4 +956,48 @@ private static ImmutableSortedMap.Builder<String, VarContext> getStringVarContex
...
+        int currentVariableVersion = GlobalVariable.variableVersion;
+        if (currentVariableVersion == GlobalVariable.VARIABLE_VERSION_0) {
+            // update from 2.0.15 or below to 2.0.16 or higher
+            if (VariableMgr.newSessionVariable().nereidsTimeoutSecond == 5) {
+                VariableMgr.refreshDefaultSessionVariables("update variable version",
+                        SessionVariable.NEREIDS_TIMEOUT_SECOND, "30");
+            }
+        }
+        if (currentVariableVersion < GlobalVariable.VARIABLE_VERSION_100) {
+            // update from 2.1.6 or below to 2.1.7 or higher
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/VariableMgr.java:970]:
```suggestion
            // update from 2.1.7 or below to 2.1.8 or higher
```

**Review by morningman — APPROVED** [review]

**Pull request merged**